src/MMALSharp.Processing/Handlers/ExternalProcessCaptureHandler.cs

```diff
@@ -0,0 +1,330 @@
+        /// and execute both of these with Task.WhenAll.
+        /// </summary>
+        /// <param name="cancellationToken">The same timeout token used for MMALCamera.ProcessAsync</param>
+        public async Task ManageProcessLifecycleAsync(CancellationToken cancellationToken)
+        {
+            var outputToken = new CancellationTokenSource();
+            
+            await Task.WhenAny(new[]
+            {
+                // this Task is the one that will be cancelled by the ProcessAsync timeout
+                cancellationToken.AsTask(),
+
+                // we control this token so this will keep running when the above expires
+                ConsoleWriteLineAsync(outputToken.Token)
+            });
+
+            // now we can do a clean shutdown; ConsoleWriteLineAsync is still running
+            if (_options.TerminationSignals.Length > 0)
+            {
+                MMALLog.Logger?.LogTrace($"Sending process termination signals");
+                foreach (var sigint in _options.TerminationSignals)
+                {
+                    if (_process.HasExited) break;
+                    Syscall.kill(_process.Id, sigint);
+                }
+            }
+
+            // gives output buffer a chance to drain, and also allows the process to do exit-cleanup
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            while (!_process.HasExited && stopwatch.ElapsedMilliseconds < _options.DrainOutputDelayMs)
+                await Task.Delay(50);
+
+            MMALLog.Logger?.LogTrace($"Process exited? {_process.HasExited}");
+
+            // now we terminate ConsoleWriteLineAsync
+            outputToken.Cancel();
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            if (!_process.HasExited)
+            {
+                MMALLog.Logger?.LogTrace($"Killing PID: {_process.Id}");
+                _process.Kill();
+            }
+
+            MMALLog.Logger?.LogTrace($"Disposing PID: {_process.Id}");
+            _process.Dispose();
+
+            MMALLog.Logger?.LogTrace("Disposed ExternalProcessCaptureHandler");
+        }
+
+        // Using "async void" is ok for an event-handler. The purpose of a Task is to communicate the
+        // result of an operation to some external "observer" -- but by definition the caller that fires
+        // an event doesn't care about the result of the operation. The only caveat is that you get no
+        // exception handling; an unhandled exception would terminate the process.
+        private async void WriteToBuffer(object sendingProcess, DataReceivedEventArgs e)
+        {
+            try
+            {
+                // Technically the faster TryWrite method is guaranteed to work for an
+                // unbounded channel, but in this case non-blocking async is more important.
+                if (_stdoutBuffer != null && e.Data != null)
```

Comment:
To keep with the styling of the rest of the project, please can you make sure that any conditional block includes curly braces.